Stats: Updating page header for summary details

grzegorz-cp opened the pull request:
#### Proposed Changes

* Updating the header for a summary details page (behind a feature flag)
* Adding a mixin file combining CSS needed by the summary details page and:
    * Section heading (used above the chart on the Traffic page)
    * Navigation segments (used above the chart on the Traffic page)

This PR continues work started in #71822 

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Navigate to `Stats` and make sure that the header above the chart on the Traffic page didn't change (consider multiple breakpoints)
* Navigate to `View all`/`View deatils` for stats modules below the chart (summary detail page) and make sure that horizontal bars are NOT visible and the original header is visible
* apply the feature flag by adding `?flags=stats/horizontal-bars-everywhere` to the URL
* confirm that the summary detail pages now show horizontal bars and a new header, similar to the one above the chart on the main page (period buttons should not have an equal width here)
* confirm that the Insights page loads without problems (The tags component is a work in progress)

Before applying a feature flag
![Screenshot 2023-01-12 at 11 36 17 AM](https://user-images.githubusercontent.com/112354940/211934790-b6517c8e-f9fe-4b90-8a22-4d29e14b6a2a.png)

Header with a feature flag not including changes in this PR
![Screenshot 2023-01-12 at 11 37 23 AM](https://user-images.githubusercontent.com/112354940/211934883-4d4dd98b-f20b-4561-9f68-e923fce611ee.png)

Header with a feature flag and proposed changes
![Screenshot 2023-01-12 at 11 37 38 AM](https://user-images.githubusercontent.com/112354940/211934964-70ee4a34-c34a-4da6-aa81-d79192a11b7e.png)


#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #71638

grzegorz-cp committed:
Stats: Updating CSS for summary details and DRYing it

grzegorz-cp commented on client/my-sites/stats/_modernized-mixins.scss:
```diff
@@ -0,0 +1,87 @@
+/**
+ * CSS used to transform existing Segment component to the styles used in the modernized Stats.
+ */
+@mixin navigation-segment-control-buttons( $segment-width: 100px ) {
```
I combined the styles in a new file. Do you think this is helpful?

grzegorz-cp commented on client/my-sites/stats/_modernized-mixins.scss:
```diff
@@ -0,0 +1,87 @@
+	}
+}
+
+// TODO: would it be better to have a mixin file in Calypso components?
```
There is a potential for improvement, including heading styles used by components we import (especially on the Insights page).

grzegorz-cp commented on client/my-sites/stats/stats-module/index.jsx:
```diff
@@ -173,12 +173,13 @@ class StatsModule extends Component {
 				) }
 				{ shouldShowNewModule && (
 					<>
+						{ /* TODO: Move this header to the summary page when modernising it */ }
```
Moving this header to the summary page will be beneficial because it will unify all of the summary pages. The only thing left is to handle `path` for all cases, so it could be a part of modernising the Summary page.

grzegorz-cp commented:
I noticed that `Videos` uses a different table, and `Filed downloads` is missing the header. I'll investigate.

jsnmoon reviewed (approved):
Changes look reasonable, though the videos metric page lacks the horizontal bars and the downloads page lacks the header, as already noted. Great work getting us on the right path to unifying all metric summary pages.

I'd like to see what the rest of the team thinks about our mixin usage here. (I like how it makes the intent of our styles more explicit.)

grzegorz-cp commented:
I found the component for the video page [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/stats/videopress-stats-module/index.jsx).
It duplicates a lot of the code used for the other modules, so there's a lot of potential to tidy it up, but it differs from all the other pages because it displays more values. Not only views but also impressions and hours watched. It doesn't have any period navigation.

grzegorz-cp commented:
The header is missing for `File downloads` and `Authors` because those modules are excluded from displaying the data by different periods. I'm not sure why. 
It makes it very difficult to check authors for different days. So at the moment, you've got to return to the traffic page, select a different day, go back to `authors` details, and so on if you'd like to check other days.

a8ck3n commented:
In my testing, it works as described in the no-feature-flag mode. I'm seeing the date header printed twice (matching your screenshot).

With the flag enabled, the header gets clipped on smaller screens. I'm seeing this:

![NewHeaderPadding](https://user-images.githubusercontent.com/40267301/212262912-588cfa98-0744-4c42-afaf-973cab562fa8.png)

If we do the same thing as on the Traffic page, we probably want the period control to drop down below the header as the screen size gets smaller.

grzegorz-cp committed:
Merge branch 'trunk' into update/stats-71638-horizontal-bar-everywhere-details

grzegorz-cp committed:
Stats: Fixing summary page header for small screens

grzegorz-cp commented:
Thanks for finding this @a8ck3n. I made several updates to fix the display:
- fixed title for various screen sizes
- added padding for CSV at the bottom of the page
- added a header variation for `Authors` and `Downloads` by keeping the new title font style but hiding the period navigation and replacing it with a CSV download button
- excluded `Tags` and `Publicize` from using new horizontal bars (they will be updated when working on the bottom part of the insights page)

More changes should probably be done as part of #71018 because it requires shifting more HTML, but if anything looks broken, I can create an issue and fix it before enabling the feature flag for production.

grzegorz-cp committed:
Merge branch 'trunk' into update/stats-71638-horizontal-bar-everywhere-details

grzegorz-cp committed:
Stats: Fixing header spacing for small screens

kangzj reviewed (approved):
It works as described and LGTM 👍 

PS: I can see the two issues pointed out by @a8ck3n are both addressed.

grzegorz-cp committed:
Merge branch 'trunk' into update/stats-71638-horizontal-bar-everywhere-details
